test/e2e-layering/onclusterbuild_test.go

```diff
@@ -225,3 +228,89 @@ func prepareForTest(t *testing.T, cs *framework.ClientSet, testOpts onClusterBui
 	_, err = helpers.WaitForRenderedConfig(t, cs, testOpts.poolName, "00-worker")
 	require.NoError(t, err)
 }
+
+func TestSSHKeyAndPasswordForOSBuilder(t *testing.T) {
+	cs := framework.NewClientSet("")
+
+	// label random node from pool, get the node
+	unlabelFunc := helpers.LabelRandomNodeFromPool(t, cs, "worker", "node-role.kubernetes.io/layered")
+	osNode := helpers.GetSingleNodeByRole(t, cs, layeredMCPName)
+
+	// prepare for on cluster build test
+	prepareForTest(t, cs, onClusterBuildTestOpts{
+		imageBuilderType:  build.OpenshiftImageBuilder,
+		poolName:          layeredMCPName,
+		customDockerfiles: map[string]string{},
+	})
+
+	// Set up Ignition config with the desired SSH key and password
+	testIgnConfig := ctrlcommon.NewIgnConfig()
+	sshKeyContent := "testsshkey11"
+	passwordHash := "testpassword11"
+
+	// retreive initial etc/shadow contents
+	initialEtcShadowContents := helpers.ExecCmdOnNode(t, cs, osNode, "grep", "^core:", "/rootfs/etc/shadow")
+
+	testIgnConfig.Passwd.Users = []ign3types.PasswdUser{
+		{
+			Name:              "core",
+			SSHAuthorizedKeys: []ign3types.SSHAuthorizedKey{ign3types.SSHAuthorizedKey(sshKeyContent)},
+			PasswordHash:      &passwordHash,
+		},
+	}
+
+	testConfig := &mcfgv1.MachineConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   "99-test-ssh-and-password",
+			Labels: helpers.MCLabelForRole("layeredo"),
```

Comment:
Change `helpers.MCLabelForRole("layeredo")` to `helpers.MCLabelForRole(layeredMCPName)` to fix the e2e-layered test failure.